### PR TITLE
RBL : add interface stm32f103xb

### PR DESCRIPTION
**Pull request by zoujixing**

### CODE REVIEW

**Comment by zoujixing** [comment]
Currently we only support target nRF51 and nRF52

**Comment by c1728p9** [comment]
Thanks for the PR.  I'll give it a try. You mention that only the nrf51 and nrf52 are supported. Do you mean the stm32f103xb interface does not work with other targets? Glancing over the code I don't see anything that is nrf51/nrf52 specific.

**Comment by zoujixing** [comment]
I just test target nRF51/nRF52. I think this support most targets, but I have not target boards to test.

**Comment by c1728p9** [comment]
Hi @zoujixing unfortunately, I haven't been able to test this as thoroughly as I would like so this won't make it into 0242. I'll try to get it in by the next release.

**Comment by zoujixing** [comment]
Hi 
    Thanks for you testing. Can you tell me where it does not pass the test？

**Comment by c1728p9** [comment]
Hi @zoujixing, currently nothing is failing. I have reviewed and built the code, but I have not had a chance to load it on a board and try it out. I would like to do this first before merging this PR.

**Comment by mfiore02** [comment]
@c1728p9 fingers crossed for this getting merged. We have a couple of DK boards for some of our mbed products and I'd love to put DAPLink on them instead of the firmware from ST. I may grab this PR and try to build firmware for our boards (Nucleo F411RE targets) - I'll let you know the results if I find the time.

File: source/hic_hal/stm32/stm32f103xb/gpio.c [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        GPIO_InitStructure.GPIO_Pin = POWER_EN_PIN;
+        GPIO_InitStructure.GPIO_Speed = GPIO_Speed_50MHz;
+        GPIO_InitStructure.GPIO_Mode = GPIO_Mode_Out_PP;
+        GPIO_Init(POWER_EN_PIN_PORT, &GPIO_InitStructure);
+        GPIO_SetBits(POWER_EN_PIN_PORT, POWER_EN_PIN);
+    }
+    // Config ReadOut protection
+    if(FLASH_GetReadOutProtectionStatus() != SET) {
+        FLASH_Unlock();
+        FLASH_ReadOutProtection(ENABLE);
```
Comment (c1728p9) [inline source/hic_hal/stm32/stm32f103xb/gpio.c]: Why is read protection enabled?

Comment (zoujixing) [inline source/hic_hal/stm32/stm32f103xb/gpio.c]: If need, I can remove it.

**Comment by c1728p9** [comment]
Hi @zoujixing, I loaded the bootloader from this PR onto a board, but I can't seem to get USB to work. Do I need any special configuration to get this to work?

**Comment by DHJeff** [comment]
Hi All
I tried to run this DAPLink FW on the STM32F103, but I can't get IDCODE from target MCU and  display an error message "RDDI-DAP Error" in the Keil, I found out the response_value and response value incorrect of DAP_SWD_Transfer function, any suggestion ? Thanks!

**Comment by devanlai** [comment]
For those interested in trying out DAPLink on Nucleo development boards, I've created a branch based off of this pull request that works with STLinkv2-1 hardware:
https://github.com/devanlai/DAPLink/tree/nucleo_f103rb

It's still pretty rough around the edges (I probably should have made a separate HIF, since the crystal frequencies are different), but it should be enough for someone to test it out on more accessible hardware.

**Comment by Cheong2K** [comment]
@devanlai, very good, please make use of this, I also hope mbed team merge this.

**Comment by neomissing** [comment]
CDC sending hundreds of bytes will crash,

**Comment by brianesquilona** [comment]
#257 was merged for stm32f103xb hic implementation

**Pull request closed without merging**